test/TEST-35-LOGIN/test.sh

```diff
@@ -8,6 +8,7 @@ TEST_DESCRIPTION="Tests for systemd-logind"
 . "${TEST_BASE_DIR:?}/test-functions"
 
 test_append_files() {
+    inst "${SOURCE_DIR}/unit-tests/manual/test-session-properties"
```

Comment:
Huh? I thought the stuff in `unit-tests/` is always installed...

Anyway, this test has been there for a year, so why would this only become a problem now?

Reply:
I guess this is a SUSE issue with `$NO_BUILD` set. If that's the case, please adjust the `install_suse_systemd()` function in `test/test-functions` instead.

Reply:
Yes it's probably due to `NO_BUILD=1` although I haven't looked at the reason why it started to fail with this setting only now.

I don't think this should be moved to `install_suse_systemd()` since `test-session-properties` is specific to TEST-35-LOGIN and the problem is not really SUSE specific but related to the NO_BUILD settings.

Reply:
Not to mention that this can't work anyway, as `test-session-properties` is a binary that needs to be built first, which doesn't happen with `$NO_BUILD` set.

Reply:
On SUSE there's also test-session-properties installed on the host (whose systemd binaries are going to be tested):

```
# rpm -ql systemd-testsuite | grep test-session
/usr/lib/systemd/tests/unit-tests/manual/test-session-properties
```
The problem is that this binary is not copied from the host to the test image. That's why I think it needs to be done in the test specific hook.

Reply:
In other cases (debian, fedora) it gets into the image since we install contents of the` systemd-tests` package. That's why I suggested to update the `install_suse_systemd()`, since you already install other test data there as SUSE apparently doesn't have the `systemd-tests` package.

Reply:
> I see but then in the case of debian or fedora for each single test (eg TEST-01-BASIC) _all_ tests and test data get installed in the image even though most of the files won't be used by the test itself, right ?

Yes. IIRC this has been done for performance reasons: this way, most tests can use the same image. But it's a moot argument anyway. The test image run by a test always contains a lot of stuff that's not needed by the test; e.g., TEST-35-LOGIN certainly doesn't need `systemd-homed` or `systemd-timesyncd`, yet both are installed.

> 
> If so I'm still not sure `install_suse_systemd()` would be the right place. IMU `test_append_files()` is the place where each test can specify their specific needs. And this is the case here, test-session-properties is only needed by TEST-35-LOGIN.

I beg to differ. IMHO `test_append_files()` should be only used to install files that are not part of systemd build. IOW, I don't see why test data and binaries should be treated differently from systemd daemons or unit files.